xarray/backends/netCDF4_.py

```diff
@@ -182,7 +183,14 @@ def _extract_nc4_variable_encoding(variable, raise_on_invalid=False,
 def _open_netcdf4_group(filename, mode, group=None, **kwargs):
     import netCDF4 as nc4
 
-    ds = nc4.Dataset(filename, mode=mode, **kwargs)
+    try:
+        ds = nc4.Dataset(filename, mode=mode, **kwargs)
+    except IOError as e:
+        message = e.args[0]
```

Comment:
I'm a little nervious about this fix -- it seems like it's correctness relies on netCDF4 raising an error with only this single argument.

Would it make sense to fix this upstream in netCDF4 instead? This looks like where the error is raised:
https://github.com/Unidata/netcdf4-python/blob/master/netCDF4/_netCDF4.pyx#L2011

Reply:
I think this is a fair point. Let's leave this open until we see if I can get this to work in netCDF4-python.